src/main/java/de/unistuttgart/iste/gits/content_service/service/ContentService.java

```diff
@@ -356,4 +370,51 @@ private List<SkillType> getSkillTypesOfChapter(final UUID chapterId) {
                 .distinct()
                 .toList();
     }
+
+
+    /**
+     * returns an ordered list of content with no links to any section for a list of chapter ID.
+     *
+     * @param chapterIds list of chapter IDs in which content is to be searched for
+     * @return ordered list of content that matches the order of the chapterID argument
+     */
+    public List<List<Content>> getContentWithNoSection(List<UUID> chapterIds) {
+        List<List<Content>> result = new ArrayList<>(chapterIds.size());
+
+        // get a list containing all sections for the given chapters, but not divided by chapter yet
+        List<SectionEntity> sectionEntities = sectionRepository.findByChapterIdIn(chapterIds);
+        List<ContentEntity> contentEntities = contentRepository.findByChapterIdIn(chapterIds);
+
+        // collect optional and required content in form of a stream for further processing
+        Stream<Set<ContentEntity>> requiredContentStream = sectionEntities.stream()
+                .map(SectionEntity::getStages)
+                .flatMap(Collection::stream)
+                .map(StageEntity::getRequiredContents);
+
+        Stream<Set<ContentEntity>> optionalContentStream = sectionEntities.stream()
+                .map(SectionEntity::getStages)
+                .flatMap(Collection::stream)
+                .map(StageEntity::getOptionalContents);
+
+        // collect content from both required and optional sets in stages
+        Set<ContentEntity> contentEntitiesInSections = Stream.concat(requiredContentStream, optionalContentStream)
+                .flatMap(Collection::stream)
+                .collect(Collectors.toSet());
+
+
+        Map<UUID, List<Content>> contentsWithNoSectionByChapterId = contentEntities.stream()
```

Comment:
consider GitsCollectionUtils.groupIntoSublists, can save you a lot of lines here